migrate.py

```diff
@@ -649,6 +690,314 @@ def copy_tests(plugin, coll, spec, args):
+            for idx, item in enumerate(value):
+                if isinstance(item, Mapping):
+                    _rewrite_yaml_mapping(el[key][idx], namespace, collection, spec)
+                else:
+                    if key == 'module_blacklist':
+                        for coll in spec.keys():
+                            if item in get_plugins_from_collection(coll, 'modules', spec):
+                                el[key][idx] = get_plugin_fqcn(namespace, coll, el[key][idx])
+                                if collection != coll:
+                                    integration_tests_add_to_deps(collection, coll)
+
+                    el[key][idx] = _rewrite_yaml_lookup(el[key][idx], namespace, collection, spec)
+                    el[key][idx] = _rewrite_yaml_filter(el[key][idx], namespace, collection, spec)
+                    el[key][idx] = _rewrite_yaml_test(el[key][idx], namespace, collection, spec)
+        elif isinstance(value, str):
+            el[key] = _rewrite_yaml_lookup(el[key], namespace, collection, spec)
+            el[key] = _rewrite_yaml_filter(el[key], namespace, collection, spec)
+            el[key] = _rewrite_yaml_test(el[key], namespace, collection, spec)
+
+
+def _rewrite_yaml_lookup(value, namespace, collection, spec):
+    if not ('lookup(' in value or 'query(' in value or 'q(' in value):
+        return value
+
+    for coll in spec.keys():
+        for plugin_name in get_plugins_from_collection(coll, 'lookup', spec):
+            if plugin_name in value:
+                value = value.replace(plugin_name, get_plugin_fqcn(namespace, coll, plugin_name))
+                if collection != coll:
+                    integration_tests_add_to_deps(collection, coll)
+
+    return value
+
+
+def _rewrite_yaml_filter(value, namespace, collection, spec):
+    if '|' not in value:
+        return value
+
+    for coll in spec.keys():
+        for filter_plugin_name in get_plugins_from_collection(coll, 'filter', spec):
+            imported_module = import_module('ansible.plugins.filter.' + filter_plugin_name)
+            fm = getattr(imported_module, 'FilterModule', None)
+            if fm is None:
+                continue
+            filters = fm().filters().keys()
+            for found_filter in (match[5] for match in FILTER_RE.findall(value)):
+                if found_filter in filters:
+                    value = value.replace(found_filter, get_plugin_fqcn(namespace, coll, found_filter))
+                    if collection != coll:
+                        integration_tests_add_to_deps(collection, coll)
+    return value
+
+
+def _rewrite_yaml_test(value, namespace, collection, spec):
+    if ' is ' not in value:
+        return value
+
+    for coll in spec.keys():
+        for test_plugin_name in get_plugins_from_collection(coll, 'test', spec):
+            imported_module = import_module('ansible.plugins.test.' + test_plugin_name)
+            tm = getattr(imported_module, 'TestModule', None)
```

Comment:
Theoretically, there could be something like `TestModule = None` inside. Thus I recommend using EAFP way:
```suggestion
            try:
                tm = getattr(imported_module, 'TestModule', None)
            except AttributeError:
                continue
```